batch/sql/add-user-resources-ready-cores-token.sql

```diff
@@ -0,0 +1,176 @@
+SET @n_tokens = 200;
+
+ALTER TABLE globals ADD COLUMN n_tokens INT NOT NULL;
+UPDATE globals SET n_tokens = @n_tokens;
+
+CREATE TABLE IF NOT EXISTS `batches_staging` (
+  `batch_id` BIGINT NOT NULL,
+  `token` INT NOT NULL,
+  `n_jobs` INT NOT NULL DEFAULT 0,
+  `n_ready_jobs` INT NOT NULL DEFAULT 0,
+  `ready_cores_mcpu` BIGINT NOT NULL DEFAULT 0,
+  PRIMARY KEY (`batch_id`, `token`),
+  FOREIGN KEY (`batch_id`) REFERENCES batches(`id`) ON DELETE CASCADE
+) ENGINE = InnoDB;
+
+ALTER TABLE user_resources ADD COLUMN token INT NOT NULL DEFAULT 0;
+ALTER TABLE user_resources DROP PRIMARY KEY, ADD PRIMARY KEY(`user`, `token`);
+
+ALTER TABLE ready_cores ADD COLUMN token INT NOT NULL DEFAULT 0;
+ALTER TABLE ready_cores ADD PRIMARY KEY(`token`);
+
+CREATE TEMPORARY TABLE tmp_resources AS (
```

Comment:
Do you need to drop a temporary table?

Reply:
No. https://dev.mysql.com/doc/refman/5.7/en/create-temporary-table.html

Reply:
Thanks!